src/H5VLcallback.c

```diff
@@ -6958,18 +7008,28 @@ H5VL__request_free(void *req, const H5VL_class_t *cls)
 herr_t
 H5VL_request_free(const H5VL_object_t *vol_obj)
 {
-    herr_t ret_value = SUCCEED; /* Return value */
+    hbool_t vol_wrapper_set = FALSE;
+    herr_t  ret_value       = SUCCEED; /* Return value */
 
     FUNC_ENTER_NOAPI(FAIL)
 
     /* Sanity check */
     HDassert(vol_obj);
 
+    /* Set wrapper info in API context */
+    if (H5VL_set_vol_wrapper(vol_obj) < 0)
+        HGOTO_ERROR(H5E_VOL, H5E_CANTSET, FAIL, "can't set VOL wrapper info")
+    vol_wrapper_set = TRUE;
```

Comment:
instead of setting those intermediate state variables, we should just have HGOTO_ERROR macros return to a different error label.